flink-core/src/test/java/org/apache/flink/util/InstantiationUtilTest.java

```diff
@@ -46,6 +50,17 @@
  */
 public class InstantiationUtilTest extends TestLogger {
 
+	@Test
+	public void testResolveProxyClass() throws Exception {
```

Comment:
This test does not check the new behavior. The proxy class is available in the same classloader as `InstantiationUtil`, which corresponds to a situation that the proxy class is available on the parent classpath in flink cluster. What you should actually test is when the proxy class is only available from user classloader. 

In `org.apache.flink.runtime.classloading.ClassLoaderTest#testMessageDecodingWithUnavailableClass` you can check how to create a new classloader with a class. Also always please make sure that your test fails without the changes that are supposed to fix the tested problem (It is not the case here, it passes without your changes).